internal/services/containers/kubernetes_cluster_trusted_access_role_binding_resource_gen.go

```diff
@@ -0,0 +1,244 @@
+package containers
+
+// NOTE: this file is generated - manual changes will be overwritten.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See NOTICE.txt in the project root for license information.
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerservice/2023-03-02-preview/trustedaccess"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+var _ sdk.Resource = KubernetesClusterTrustedAccessRoleBindingResource{}
+var _ sdk.ResourceWithUpdate = KubernetesClusterTrustedAccessRoleBindingResource{}
+
+type KubernetesClusterTrustedAccessRoleBindingResource struct{}
+
+func (r KubernetesClusterTrustedAccessRoleBindingResource) ModelObject() interface{} {
+	return &KubernetesClusterTrustedAccessRoleBindingResourceSchema{}
+}
+
+type KubernetesClusterTrustedAccessRoleBindingResourceSchema struct {
+	KubernetesClusterId string   `tfschema:"kubernetes_cluster_id"`
+	Name                string   `tfschema:"name"`
+	Roles               []string `tfschema:"roles"`
+	SourceResourceId    string   `tfschema:"source_resource_id"`
+}
+
+func (r KubernetesClusterTrustedAccessRoleBindingResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return trustedaccess.ValidateTrustedAccessRoleBindingID
+}
+func (r KubernetesClusterTrustedAccessRoleBindingResource) ResourceType() string {
+	return "azurerm_kubernetes_cluster_trusted_access_role_binding"
+}
+func (r KubernetesClusterTrustedAccessRoleBindingResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"kubernetes_cluster_id": {
+			ForceNew: true,
+			Required: true,
+			Type:     pluginsdk.TypeString,
```

Comment:
since we know what the Parent Resource ID is, can we add validation to check this is correct?